var/spack/repos/builtin/packages/snakemake/package.py

```diff
@@ -54,38 +56,53 @@ class Snakemake(PythonPackage):
     depends_on("py-nbformat", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"), when="@7.29.0:")
     depends_on("py-psutil", type=("build", "run"))
-    depends_on("py-pulp@2:", type=("build", "run"))
+    depends_on("py-pulp@2.3.1:2.8", type=("build", "run"), when="@8.1.2:")
+    depends_on("py-pulp@2:", type=("build", "run"), when="@:8.1.1")
     depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-requests@2.8.1:2", type=("build", "run"), when="@8.4.12")
     depends_on("py-requests", type=("build", "run"))
     depends_on("py-reretry", type=("build", "run"), when="@7:")
+    depends_on("py-smart-open@3:6", type=("build", "run"), when="@8.4.12:")
     depends_on("py-smart-open@3:", type=("build", "run"))
+    depends_on(
+        "py-snakemake-interface-executor-plugins@8.1.3:8", type=("build", "run"), when="@8:"
+    )
+    depends_on("py-snakemake-interface-common@1.17:1", type=("build", "run"), when="@8.4.10:")
+    depends_on("py-snakemake-interface-common@1.15:1", type=("build", "run"), when="@8:")
+    depends_on(
+        "py-snakemake-interface-storage-plugins@3.1:3", type=("build", "run"), when="@8.4.10:"
+    )
+    depends_on("py-snakemake-interface-storage-plugins@3", type=("build", "run"), when="@8:")
     depends_on("py-stopit", type=("build", "run"))
     depends_on("py-tabulate", type=("build", "run"))
     depends_on("py-throttler", type=("build", "run"), when="@7:")
+    depends_on("py-toposort@1.10:1", type=("build", "run"), when="@8.4.12:")
     depends_on("py-toposort@1.10:", type=("build", "run"), when="@7.24.0:")
     depends_on("py-toposort", type=("build", "run"), when="@:7.23")
     depends_on("py-wrapt", type=("build", "run"))
     depends_on("py-yte@1.5.1:1", type=("build", "run"), when="@7.28.1:")
     depends_on("py-yte@1", type=("build", "run"), when="@7:7.28.0")
+    depends_on("py-dpath@2.1.6:2", type=("build", "run"), when="@8.3:")
+    depends_on("py-conda-inject@1.3.1:1", type=("build", "run"), when="@8:")
 
     variant("reports", default=False, description="Generate self-contained HTML reports")
 
     with when("+reports"):
         depends_on("py-pygments", type=("build", "run"))
-        # Historical dependencies
```

Comment:
I usually keep comments like this to remind me I can skip these deps during review

Reply:
I have added this in the previous pull request, but I think it doesn't help this package, so removing it now.

There is a dozen or two of other past dependencies which are not marked, so this just adds noise.